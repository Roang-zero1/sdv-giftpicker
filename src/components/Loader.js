import React, { Component } from 'react';
import { Container } from 'reactstrap';
import './Loader.css';

class Loader extends Component {
  render() {
    return (
      <Container className={'loader-wrapper'}>
        <h2>Loading</h2>
        <p>Loading Data please wait</p>
        <div className={'loader'} />
      </Container>
    );
  }
}

export default Loader;
