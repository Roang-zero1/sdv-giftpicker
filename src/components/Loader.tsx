import * as React from 'react';
import { Component } from 'react';

import { Container } from 'reactstrap';
import styled from 'styled-components';

const Wrapper = styled(Container)`
  height: 14em;
`;

const LoaderDiv = styled.div`
  direction: ltr;
  text-indent: -9999em;
  margin: 3em auto;
  position: relative;
  font-size: 11px;
  transform: translateZ(0);

  &,
  &::after,
  &::before {
    background: #222222;
    animation: load1 1s infinite ease-in-out;
    animation-delay: -160ms;
    width: 1em;
    height: 4em;
  }

  &::before,
  &::after {
    position: absolute;
    top: 0;
    content: '';
  }

  &::before {
    left: -1.5em;
    animation-delay: -320ms;
  }

  &::after {
    left: 1.5em;
    animation-delay: 0ms;
  }
  @keyframes load1 {
    0%,
    80%,
    100% {
      box-shadow: 0 0;
      height: 4em;
    }
    40% {
      box-shadow: 0 -2em;
      height: 5em;
    }
  }
`;

class Loader extends Component {
  public render() {
    return (
      <Wrapper>
        <h2>Loading</h2>
        <p>Loading Data please wait</p>
        <LoaderDiv />
      </Wrapper>
    );
  }
}

export default Loader;
