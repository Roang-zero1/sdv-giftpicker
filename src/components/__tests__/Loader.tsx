import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import * as React from 'react';
import Loader from '../Loader';

describe('components/Loader --- Shallow render component', () => {
  let cut: ShallowWrapper;

  beforeEach(() => {
    cut = shallow(<Loader />);
  });

  it('should render the one component', () => {
    expect(cut.length).toEqual(1);
  });

  it('should be the same as the last snapshot', () => {
    expect(cut).toMatchSnapshot();
  });
});

describe('components/Loader --- Render component', () => {
  let cut: ReactWrapper;

  beforeEach(() => {
    cut = mount(<Loader />);
  });

  it('should render the one component', () => {
    expect(cut.length).toEqual(1);
  });

  it('should be the same as the last snapshot', () => {
    expect(cut).toMatchSnapshot();
  });
});
