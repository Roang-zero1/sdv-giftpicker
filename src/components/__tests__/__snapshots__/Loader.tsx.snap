// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Loader --- Render component should be the same as the last snapshot 1`] = `
.c0 {
  height: 14em;
}

.c1 {
  direction: ltr;
  text-indent: -9999em;
  margin: 3em auto;
  position: relative;
  font-size: 11px;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c1,
.c1::after,
.c1::before {
  background: #222222;
  -webkit-animation: load1 1s infinite ease-in-out;
  animation: load1 1s infinite ease-in-out;
  -webkit-animation-delay: -160ms;
  animation-delay: -160ms;
  width: 1em;
  height: 4em;
}

.c1::before,
.c1::after {
  position: absolute;
  top: 0;
  content: '';
}

.c1::before {
  left: -1.5em;
  -webkit-animation-delay: -320ms;
  animation-delay: -320ms;
}

.c1::after {
  left: 1.5em;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

<Loader>
  <Styled(Container)>
    <Container
      className="c0"
      tag="div"
    >
      <div
        className="c0 container"
      >
        <h2>
          Loading
        </h2>
        <p>
          Loading Data please wait
        </p>
        <styled.div>
          <div
            className="c1"
          />
        </styled.div>
      </div>
    </Container>
  </Styled(Container)>
</Loader>
`;

exports[`components/Loader --- Shallow render component should be the same as the last snapshot 1`] = `
<Styled(Container)>
  <h2>
    Loading
  </h2>
  <p>
    Loading Data please wait
  </p>
  <styled.div />
</Styled(Container)>
`;
